packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -1090,6 +1094,216 @@ describe("SharedTree", () => {
+			 * (C) Commit 1: restore the removed tree, Commit 2: make an edit to the restored tree
+			 * Peer 2 can only do option (A) above.
+			 */
+			const scenarios = [
+				["A", "A"],
+				["A", "B"],
+				["A", "C"],
+				["B", "A"],
+				["C", "A"],
+			];
+
+			function makeTestTitle(peer: string): string {
+				switch (peer) {
+					case "A": {
+						return "an edit to the removed tree after another edit";
+					}
+					case "B": {
+						return "the restoration of a tree after an edit to the tree";
+					}
+					case "C": {
+						return "the edit to a removed tree after its restoration";
+					}
+					default:
+						fail("invalid test case");
+				}
+			}
+
+			function makeEdits(
+				peer: CheckoutFlexTreeView<typeof schema.rootFieldSchema>,
+				edits: string,
+			): Revertible[] {
+				const undos: Revertible[] = [];
+				const unsubscribe = peer.checkout.events.on(
+					"newRevertible",
+					(revertible: Revertible) => {
+						if (revertible.kind !== RevertibleKind.Undo) {
+							revertible.retain();
+							undos.push(revertible);
+						}
+					},
+				);
+
+				if (edits === "A") {
+					makeAEdits(peer);
+				} else if (edits === "B" || edits === "C") {
+					makeBOrCEdits(peer);
+				} else {
+					fail("invalid test scenario");
+				}
+
+				unsubscribe();
+				return undos;
+			}
+
+			function makeAEdits(peer: CheckoutFlexTreeView<typeof schema.rootFieldSchema>): void {
+				const outerList = peer.flexTree.content.content;
+				const innerList = (outerList.at(0) ?? assert.fail()).content;
+				innerList.insertAtEnd("b");
+				innerList.insertAtEnd("c");
+			}
+
+			function makeBOrCEdits(
+				peer: CheckoutFlexTreeView<typeof schema.rootFieldSchema>,
+			): void {
+				const outerList = peer.flexTree.content.content;
+				const innerList = (outerList.at(0) ?? assert.fail()).content;
+				innerList.insertAtEnd("d");
+				peer.flexTree.content.content.removeAt(0);
+			}
+
+			scenarios.forEach(([otherPeer, resubmitPeer]) => {
+				it(`${makeTestTitle(resubmitPeer)} after ${makeTestTitle(
+					otherPeer,
+				)} on another peer`, () => {
+					const provider = new TestTreeProviderLite(2);
+					const content = {
+						schema,
+						allowedSchemaModifications: AllowedUpdateType.None,
+						initialTree: [["a"]],
+					};
+					const resubmitTree = provider.trees[0].schematizeInternal(content);
+					const otherTree = provider.trees[1].schematizeInternal(content);
+					provider.processMessages();
+
+					// validate insertion
+					validateTreeContent(otherTree.checkout, content);
+
+					const otherUndos = makeEdits(otherTree, otherPeer);
+					const resubmitUndos = makeEdits(resubmitTree, resubmitPeer);
+
+					provider.processMessages();
+
+					if (otherPeer === "A" && resubmitPeer === "A") {
+						assert.deepEqual(
+							// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+							[...otherTree.flexTree.content.content.at(0)!.content],
```

Comment:
`*Tree.flexTree.content.content` and `*Tree.flexTree.content` seem to get repeated a lot. Could we grab the inner and outer lists for each peer at the start of the test instead of walking down every time? I expect that's more likely to be how our users would do it.